### DISCUSSION TITLE
Allow using existing layouts in custom extensions

### DISCUSSION THREAD

**Discussion by Guiqft** [directus/directus#7063] [Feature Requests]
Hi, I'm on rc.86 and trying to use the <layout-cards /> component inside a interface and I'm getting the following error:

`[useLayoutState]: This function has to be used inside a layout component.`

When on Vue2 with rc.74, I could use the component without problems. 
What I need to do?

**Comment by rijkvanzanten** [comment]
What did you try currently?

cc @nickrum

**Reply by Guiqft** [reply]
Just trying to use `<layout-cards  collection="collection_name" :selection="selection" />` inside a interface.
I wanna achieve the same behavior of `<collection-drawer />` , being able to select collection items on a drawer component.

**Reply by Guiqft** [reply]
When on Vue2 I could normally use `<layout-tabular />, <layout-cards />`, etc.
Since Vue3, I'm getting the error from [use-layout-state.ts](https://github.com/directus/directus/blob/main/packages/shared/src/composables/use-layout-state.ts)

**Reply by nickrum** [reply]
Currently, it is not possible to directly use layouts inside extensions as they now consist of multiple components and have to be initialized by using the `useLayout()` composable which is not exposed to extensions.
I can see how this is a valid use case. We have to either expose the `useLayout()` composable (which might be tricky as it has to have access to all registered layouts) or provide a dedicated composable to pass state to the main layout components.

**Comment by nickrum** [comment]
`useLayout()` has since been exposed to extensions.

**Reply by michaeltoohig** [reply]
@nickrum Do you have any docs on how to use it? I'm a bit lost going through the source code.

Am I wrong to believe I can use `useLayout` to create a `tabular` layout found as the default in the Content modules? I want to display a custom table of data and I think this is the direction I should go for my own module.
